platformio.ini

```diff
@@ -1426,6 +1461,26 @@ extra_scripts        = ${common.extra_scripts}
   pre:buildroot/share/PlatformIO/scripts/generic_create_variant.py
   buildroot/share/PlatformIO/scripts/stm32_bootloader.py
 
+#
+# MKS Robin Nano V3 with USB Flash Drive Support
+# Currently, using a STM32duino fork, until USB Host get merged
+#
+[env:mks_robin_nano_v3_usb_flash_drive]
+extends              = env:mks_robin_nano_v3
+platform_packages    = framework-arduinoststm32@https://github.com/rhapsodyv/Arduino_Core_STM32/archive/usb-host-msc.zip
+build_flags          = ${common_stm32.build_flags}
+  -DHAL_PCD_MODULE_ENABLED
+  -DUSBCON
+  -DUSBHOST 
+  -DHAL_HCD_MODULE_ENABLED
+  -DUSE_USBHOST_HS
+  -DUSBH_IRQ_PRIO=3
+  -DUSBH_IRQ_SUBPRIO=4
+  -DUSBD_IRQ_PRIO=5
+  -DUSBD_IRQ_SUBPRIO=6
+  -DUSE_USB_HS_IN_FS
+  -DUSBD_USE_CDC
```

Comment:
Is using USB Host Drive support always going to require separate PIO environments for all the defines?

I don't like the idea of each board having a bunch of different environments...that scales very poorly as more boards use the features.

Reply:
I just separated it because of the framework... I didn't want to change the framework for the "default" env of each board.

Reply:
It would be nice to "mix in" the needed configuration options based on the users configuration/pins. USB will end up being some mixture of CDC/Composite/MassStorage/OTG/Host. If we can find a way to define those as config options perhaps the dependency-detection script can add the proper build_flags for them.

One concern about getting very clever with build flags is that VS Code doesn't seem to handle them well for syntax highlighting and contexpr evaluation. I have noticed that when I add build_flags inside a pins file that I do not see their influence inside the editor (they build fine, though).

Reply:
We can use `BOARD_CUSTOM_BUILD_FLAGS`, but if it will make hard to vs code handle them, I think we should keep doing the "old" way.. :-/
We already have too much macros for a free editor handle... now if we start adding macros that add macros, it will be crazy fast enough :-p

Reply:
> VS Code doesn't seem to handle them well for syntax highlighting and constexpr evaluation

Marlin's codebase isn't ever going to very friendly to VSCode's built-in C++ code evaluation or IntelliSense. Most of the time I use a text editor that only has "naive" syntax highlighting, and that is usually all I need. But it would be nice to have the syntax highlighting work in the debugger, to make it easier to skip past unused code.

>  I think we should keep doing the "old" way.. :-/

To make the code slightly less redundant I added a 'mix-in' environment with the new platform and flags.

Reply:
VS Code actually does a pretty good job, even with all Marlin's weirdness. It won't highlight inside `TERN` macros, but seems to be ok with just about everything else.

I rely quite a bit on using constexpr variables to validate things in the editor, without having to actually put them on hardware or even in a simulator.

I think we should be striving to help our tools work better, rather than making it harder.

I don't know how the "mix-in" will work. I was thinking something along those lines as I was looking at it, but didn't actually try it.